wgpu-core/src/device/global.rs

```diff
@@ -985,6 +985,46 @@ impl Global {
         (id, Some(error))
     }
 
+    // Unsafe-ness of internal calls has little to do with unsafe-ness of this.
+    #[allow(unused_unsafe)]
+    /// # Safety
+    ///
+    /// This function passes MSL source  code to the backend as-is and can potentially result in a
+    /// driver crash.
+    pub unsafe fn device_create_shader_module_msl(
```

Comment:
I think this combined api should go all the way through to wgpu-hal. There's no real reason to duplicate the endpoints as all, as all stages are "passing straight through" until you get to wgpu-hal. The only thing would be to have wgpu-core validate that you're passing the correct variant for the backend. This can be done with a match statement and doesn't need separate entry points however.